Step3-kakao-tech-campus/Team3_BE
# 4주차 개발 내역 정리

## Summary

4주차( + 3주차)에 개발한 내역을 정리하여 병합합니다.

## Description

- 프로젝트를 생성하고 security 설정 및 인증 필터를 생성하였습니다.
- GET api들에 대해 가짜 데이터를 주는 mock api를 작성해두었습니다.
- mock api와 별개로 이번 주까지 구현한 api 목록은 다음과 같습니다.
  - POST /api/join
  - POST /api/login
  - POST /api/logout
  - POST /api/authentication
  - POST /api/email-verification
  - POST /api/email-confirm
  - GET /api/cities
  - GET /api/cites/{cityId}/countries
  - GET /api/cities/countryId}/districts
  - GET /api/posts/{postId}
  - GET /api/posts
  - POST /api/posts
  - PUT /api/posts/{postId}
  - DELETE /api/posts/{postId}
  - GET /api/posts/{postId}/applicants
  - POST /api/posts/{postId}/applicants
  - PUT /api/posts/{postId}/applicants/{applicantId}
  - DELETE /api/posts/{postId}/applicants/{applicantId}
- jwt 인증의 경우 access token과 redis를 활용한 refresh token으로 구현되었습니다.
  - RTR(Refresh Token Rotation) 전략으로 refresh token을 1회성으로 만들었습니다.
  - 인증을 위해 POST /api/authentication으로 요청이 오고 정상 응답이 돌아가면 access와 refresh가 재발급되어 돌려줍니다.
- mysql과 redis의 경우 docker-compose를 사용하여 docker container로 띄웠습니다.
- 이메일 인증 구현을 위해 gmail의 smtp 서비스를 활용하였습니다.
- 행정구역 정보를 서버의 db에 저장해서 사용하기 위해 [행정표준코드관리시스템 - 법정동코드목록조회](https://www.code.go.kr/stdcode/regCodeL.do)의 데이터를 파싱하여 db에 저장해두었습니다.
- 프론트 쪽으로의 api 전달을 위해 개인 aws 서버를 사용하여 배포 중입니다.
  - https://server.jagaldol.dev:8080 에서 현재까지 만들어진 api를 사용할 수 있습니다.
